### PR TITLE
api/server/router/container: minor nits in exec router

### PR DESCRIPTION
**Pull request by thaJeztah**
- remove intermediate variable
- format a "todo" comment as an actual todo ':)
- explicitly suppress some unhandled errors to keep linters happy

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: api/server/router/container/exec.go [hunk 1]
```diff
@@ -114,39 +114,40 @@ func (s *containerRouter) postContainerExecStart(ctx context.Context, w http.Res
...
 			stderr = stdcopy.NewStdWriter(outStream, stdcopy.Stderr)
 			stdout = stdcopy.NewStdWriter(outStream, stdcopy.Stdout)
 		}
 	}
 
-	options := container.ExecStartOptions{
+	// Now run the user process in container.
+	//
+	// TODO: Maybe we should we pass ctx here if we're not detaching?
+	err := s.backend.ContainerExecStart(context.Background(), execName, container.ExecStartOptions{
```
Comment (thaJeztah) [inline api/server/router/container/exec.go:140]: Looks like something we could indeed consider?

File: api/server/router/container/exec.go [hunk 2]
```diff
@@ -114,39 +114,40 @@ func (s *containerRouter) postContainerExecStart(ctx context.Context, w http.Res
...
-	// Now run the user process in container.
-	// Maybe we should we pass ctx here if we're not detaching?
-	if err := s.backend.ContainerExecStart(context.Background(), execName, options); err != nil {
+	})
+	if err != nil {
 		if execStartCheck.Detach {
 			return err
 		}
-		stdout.Write([]byte(err.Error() + "\r\n"))
+		_, _ = fmt.Fprintf(stdout, "%v\r\n", err)
```
Comment (thaJeztah) [inline api/server/router/container/exec.go:150]: LOL, my IDE had a bit of a false positive here; it kept complaining "err can be nil!" (not taking the `if err != nil` check above into account. 

As all other code was using `fmt.Fprintf`, I decided to just go with that to also make it more consistent.

Comment (thaJeztah) [reply api/server/router/container/exec.go:150]: Hm... wondering if this relates to the failure;

```
=== Failed
=== FAIL: amd64.integration-cli TestDockerCLIExecSuite/TestExecInteractive (0.34s)
    docker_cli_exec_test.go:64: assertion failed: error is not nil: EOF
    --- FAIL: TestDockerCLIExecSuite/TestExecInteractive (0.34s)
```

**Comment by thaJeztah** [comment]
Weird; changed that one line back, but still getting a failure; can't see immediately what would break this though

```
=== Failed
=== FAIL: amd64.integration-cli TestDockerCLIExecSuite/TestExecInteractive (0.35s)
    docker_cli_exec_test.go:64: assertion failed: error is not nil: EOF
    --- FAIL: TestDockerCLIExecSuite/TestExecInteractive (0.35s)
```

**Review by vvoland — APPROVED** [review]

File: api/server/router/container/exec.go (hunk 1 above)
Comment (Benehiko) [reply api/server/router/container/exec.go:140]: wouldn't we also want to pass in the ctx from the function parameter to keep data for tracing? I'm not too familiar with this repository but I'd we could get better value from `context.WithoutCancel(ctx)`.

Comment (thaJeztah) [reply api/server/router/container/exec.go:140]: Yup, quite likely. I left that for now, as I only did some cleanup for some other changes I was working on, but we should definitely look.

**Pull request merged**